### PR TITLE
Sergkanz/eventsource

### PR DESCRIPTION
**Pull request by SergeyKanzhelev**
Use EventSource via Dependency Injection. One example of how to read event source messages. However this example is not working as TestExplorer doesn't support test output today. 

@upendras  - who was a contact in Visual Studio to ask when they'll support test output in Test Explorer?

### CODE REVIEW

**Comment by upendras** [comment]
@danroth27 or @vijayrkn can comment on Test Explorer

File: src/Microsoft.ApplicationInsights.AspNet/TelemetryInitializers/WebSessionTelemetryInitializer.cs [hunk 1]
```diff
@@ -33,11 +37,12 @@ protected override void OnInitializeTelemetry(HttpContext platformContext, Reque
             }
         }
 
-        private static void UpdateRequestTelemetryFromPlatformContext(RequestTelemetry requestTelemetry, HttpContext platformContext)
+        private void UpdateRequestTelemetryFromPlatformContext(RequestTelemetry requestTelemetry, HttpContext platformContext)
         {
             if (platformContext.Request.Cookies != null && platformContext.Request.Cookies.ContainsKey(WebSessionCookieName))
             {
                 var sessionCookieValue = platformContext.Request.Cookies[WebSessionCookieName];
+                bool cookieWasRead = false;
```
Comment (upendras) [inline src/Microsoft.ApplicationInsights.AspNet/TelemetryInitializers/WebSessionTelemetryInitializer.cs]: where does cookieWasRead get set to true?

**Comment by upendras** [comment]
:shipit:

File: src/Microsoft.ApplicationInsights.AspNet/Tracing/Aspnet5EventSource.cs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+    [EventSource(Name = "Microsoft-ApplicationInsights-AspNet5")]
+    public class AspNet5EventSource : EventSource
+    {
+        public AspNet5EventSource()
+        {
+        }
+
+        [Event(1, Message = "Telemetry Initializer will not collect data when HttpContext is null.", Level = EventLevel.Verbose)]
+        public void VerboseTelemetryInitializerNotEnabledOnHttpContextNull()
```
Comment (olegsych) [inline src/Microsoft.ApplicationInsights.AspNet/Tracing/Aspnet5EventSource.cs]: Why do we want to include level (error, verbose) and keyword (UserActionable) in the names of logging methods?

File: test/FunctionalTestUtils/Aspnet5EventTestListener.cs [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+            }
+        }
+
+        public string Issues
+        {
+            get
+            {
+                string issues = string.Empty;
+
+                Parallel.ForEach(this.Events.Where(issueDefinition), (message) => { issues += message.Message + Environment.NewLine; });
```
Comment (olegsych) [inline test/FunctionalTestUtils/Aspnet5EventTestListener.cs]: Is manipulation of `issues` in `Parallel.ForEach` thread-safe?

File: src/Microsoft.ApplicationInsights.AspNet/TelemetryInitializers/UserAgentTelemetryInitializer.cs [hunk 4]
```diff
@@ -14,7 +16,8 @@
     /// </summary>
     public class UserAgentTelemetryInitializer : TelemetryInitializerBase
     {
-        public UserAgentTelemetryInitializer(IHttpContextAccessor httpContextAccessor) : base(httpContextAccessor)
+        public UserAgentTelemetryInitializer(IHttpContextAccessor httpContextAccessor, AspNet5EventSource eventSource)
```
Comment (olegsych) [inline src/Microsoft.ApplicationInsights.AspNet/TelemetryInitializers/UserAgentTelemetryInitializer.cs]: Have we decided _not_ to use ASPNET's `ILogger` in favor of `EventSource`? Are we _sure_ we want a dependency on the concrete `AspNet5EventSource` class as opposed to an interface that the class implements?

**Comment by olegsych** [comment]
No tests for diagnostics code? :sob:

File: test/FunctionalTestUtils/Aspnet5EventTestListener.cs (hunk 3 above)
Comment (ChadLevy) [inline test/FunctionalTestUtils/Aspnet5EventTestListener.cs]:
```c#
var issues = new ConcurrentBag<string>();

Parallel.ForEach(this.Events.Where(issueDefinition), (message) => { issues.Add(message.Message); });

return string.Join(Environment.NewLine, issues);

```

**Pull request closed without merging**